SPECS/haproxy.spec

```diff
@@ -54,6 +54,13 @@ Requires(preun):    systemd
 Requires(postun):   systemd
 %endif
 
+%if 0%{?amzn2023}
+BuildRequires:      systemd-devel
```

Comment:
Seems like `systemd-units` is not available on Amazon Linux 2023, removed it from `BuildRequires` block